transcom/mymove
# B 18885 sit delivery miles calculation int

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A907752)

## Summary

We want to calculate and store the delivery miles value between the following for SIT service items:

On creation:

Destination SIT: distance between shipment destination address & service item destination address
Origin SIT: distance between shipment pickup address & service item pickup address

On update:

Destination SIT: distance between shipment destination address & service item ORIGINAL destination address
Origin SIT: distance between shipment pickup address & service item ORIGINAL pickup address

The service item does not have the original_address saved until the TOO approves it, so on creation it will always be 1 mile since the shipment destination/pickup address is the same as the service item’s.

Where calculations are done for:
Origin Creation: `calculateSITDeliveryMiles`
Origin Update: TOO approval of service item: `updateServiceItem`
Origin Update: prime edits pickup address: `UpdateOriginSITServiceItemSITDeliveryMiles`

Destination Creation: `calculateSITDeliveryMiles`
Destination Update: TOO approval of of service item: `updateServiceItem`
Destination Update: prime makes address update request, TOO rejects/approves: `UpdateDestinationSITServiceItemsSITDeliveryMiles` 

To test:
See the document I attached to the story as a guide.

1. Keep track of your zip codes as you go, especially the starting pickup and destination zips.
2. On a shipment, request origin SIT. 
3. Verify that SITDeliveryMiles is 1 for origin service items.
4. Approve as the TOO and check that the SITDeliveryMiles makes sense. It will likely still be 1.
5. Edit the shipment pickup address as the prime. I recommend looking at the functions for each situation and attaching a breakpoint to make sure the zips compared make sense.
6. On any shipment, doesn't have to be the same one, request destination SIT.
7. Verify that SITDeliveryMiles is 1 for destination service items.
8. Approve as the TOO and check that the SITDeliveryMiles makes sense.
9. As the prime, request an address change that does not trigger TOO approval (less than 50 miles) via the "Update Shipment Destination Address" button.
10. Verify that the SITDeliveryMiles calculation makes sense and the database value is updated to match the calculation, calc done in UpdateDestinationSITServiceItemsSITDeliveryMiles function.
11. As the prime, request an address change that triggers TOO approval via the "Update Shipment Destination Address" button.
12. As the TOO, reject it. Verify that the SITDeliveryMiles is the same as it was before rejecting the address update.
13. As the prime, request an address change that triggers TOO approval via the "Update Shipment Destination Address" button.
14. As the TOO, approve it. Verify that the SITDeliveryMiles calculation makes sense for the provided zips and the database value is updated the match the calculation.


For destination address changes that don't trigger TOO approval, you can use the BILWEI move and change destination address zip from 94535 to 94533, and should see 6. You can change it to 94534 and should see 8 miles.

For destination address changes that require TOO approval, the SITDeliveryMiles should be the same as the distance shown to the TOO when they have to approve it. As soon as the TOO approves the address change, the database should update.

Spoke with Danny and he clarified that SITDeliveryMiles should only be calculated for origin DOPSIT and DOSFSC and destination DDDSIT and DDSFSC. That update has been made.
